src/waveform/widgets/waveformwidgetinfo.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "waveform/widgets/waveformwidgetcategory.h"
+#include "waveform/widgets/waveformwidgettype.h"
+
+class WaveformWidgetAbstract;
+
+class WaveformWidgetInfoBase {
+  public:
+    const WaveformWidgetType::Type m_type;
+    const QString m_name;
+    const bool m_useGL;
+    const bool m_useGLES;
+    const bool m_useGLSL;
+    const bool m_highDetail;
+    const WaveformWidgetCategory m_category;
+
+    WaveformWidgetInfoBase(WaveformWidgetType::Type type,
+            QString name,
+            bool useGL,
+            bool useGLES,
+            bool useGLSL,
+            bool highDetail,
+            WaveformWidgetCategory category)
```

Comment:
instead of an opaque constructor with a lot of bools, it would be clearer for this take a flag enum or bitset.  or maybe this sweet new c++20 feature? https://www.cppstories.com/2017/03/on-toggle-parameters/#how-about-c20

Reply:
Yes, it most definitely would, but I prefer to do these refactorings one step at a time. The original code uses these booleans, so I will stay close to that approach in this PR. A follow up PR can be uses to change them for flags.